Source/DiabloUI/multi/selgame.cpp

```diff
@@ -541,6 +542,82 @@ void selgame_Speed_Select(size_t value)
+	    _("Create Game").data(), rect1, UiFlags::AlignCenter | UiFlags::FontSize30 | UiFlags::ColorUiSilver, 3));
+
+	SDL_Rect rect2 = { (Sint16)(uiPosition.x + 35), (Sint16)(uiPosition.y + 211), 205, 192 };
+	vecSelGameDialog.push_back(std::make_unique<UiArtText>(_("Description:").data(), rect2, UiFlags::FontSize24 | UiFlags::ColorUiSilver));
+
+	CopyUtf8(selgame_Description, _("Setting a game seed allows you to start a game with a specific map layout and item drops. Leave it blank for a random seed."), sizeof(selgame_Description));
+	CopyUtf8(selgame_Description, WordWrapString(selgame_Description, DESCRIPTION_WIDTH), sizeof(selgame_Description));
+
+	SDL_Rect rect3 = { (Sint16)(uiPosition.x + 35), (Sint16)(uiPosition.y + 256), DESCRIPTION_WIDTH, 192 };
+	vecSelGameDialog.push_back(std::make_unique<UiArtText>(selgame_Description, rect3, UiFlags::FontSize12 | UiFlags::ColorUiSilverDark, 1, 16));
+
+	auto currentTime = static_cast<uint32_t>(time(nullptr));
+	std::snprintf(selgame_Seed, sizeof(selgame_Seed), "%d", currentTime);
+
+	SDL_Rect rect4 = { (Sint16)(uiPosition.x + 305), (Sint16)(uiPosition.y + 211), 285, 33 };
+	vecSelGameDialog.push_back(std::make_unique<UiArtText>(
+	    _("Enter Game Seed").data(), rect4, UiFlags::AlignCenter | UiFlags::FontSize30 | UiFlags::ColorUiSilver, 3));
+
+	SDL_Rect rect5 = { (Sint16)(uiPosition.x + 305), (Sint16)(uiPosition.y + 314), 285, 33 };
+	vecSelGameDialog.push_back(std::make_unique<UiEdit>(
+	    "", selgame_Seed, 11, true, rect5, UiFlags::FontSize24 | UiFlags::ColorUiGold));
+
+	SDL_Rect rect6 = { (Sint16)(uiPosition.x + 299), (Sint16)(uiPosition.y + 427), 140, 35 };
+	vecSelGameDialog.push_back(std::make_unique<UiArtTextButton>(
+	    _("OK").data(),
+	    UiFocusNavigationSelect,
+	    rect6,
+	    UiFlags::AlignCenter | UiFlags::VerticalCenter | UiFlags::FontSize30 | UiFlags::ColorUiGold));
+
+	SDL_Rect rect7 = { (Sint16)(uiPosition.x + 449), (Sint16)(uiPosition.y + 427), 140, 35 };
+	vecSelGameDialog.push_back(std::make_unique<UiArtTextButton>(
+	    _("CANCEL").data(),
+	    UiFocusNavigationEsc,
+	    rect7,
+	    UiFlags::AlignCenter | UiFlags::VerticalCenter | UiFlags::FontSize30 | UiFlags::ColorUiGold));
+
+	UiInitList(nullptr, selgame_Seed_Select, selgame_Seed_Esc, vecSelGameDialog);
+}
+
+void selgame_Seed_Select(size_t /*value*/)
+{
+	// Convert seed input to an integer, ensuring it's valid
+	char *end;
+	long seed = std::strtol(selgame_Seed, &end, 10);
+
+	// Handle edge cases:
+	if (end == selgame_Seed || *end != '\0') {
+		// Invalid input, treat as blank
+		seed = -1;
+	} else if (seed < -1 || seed >= 2147483648) { // Handle the 2038 problem
+		UiSelOkDialog(_("Invalid Seed").data(), _("Please enter a valid seed (0 - 2147483647) or leave it blank.").data(), false);
+		selgame_Init();
+		UiInitGameSelectionList("");
+		return;
+	}
+
+	if (seed == -1) {
+		sgGameInitInfo.dwSeed = static_cast<uint32_t>(time(nullptr));
+	} else {
+		sgGameInitInfo.dwSeed = static_cast<uint32_t>(seed);
+	}

```

Comment:
You can simplify this by using `ParseInt()`. That function is a wrapper for `std::from_chars` and will return `ParseIntError::ParseError` for non-numeric inputs (the cases you want to treat as blank), `ParseIntError::OutOfRange` for values outside the typical range the vanilla game would use, or a value of whatever type you specify on success.

You're cutting out some valid seeds here too. seeds 0 to 315532800 are between 1970 and dec 31 1979 (hard to reach on some systems due to DOS restrictions). seeds 315532801 to 2177452799 are 1980-dec 31 2038, there's an overflow period between the 2038 epoch and the end of the year that leads to diablo still getting unique values. -1 then kicks in for dates from 01 jan 2039 onward (technically making that a valid vanilla seed too).

You probably want `ParseInt<int64_t>(selgame_Seed, -1, 2177452799)` to catch all possible cases.

[staphen's pr introducing a new rng and using it for dungeon seeds](https://github.com/diasurgical/devilutionX/pull/7030) will require adjustments here to allow the user to provide a longer seed or choose to use the old dungeon seed generation logic.